server/src/bin/delete-worker.rs

```diff
@@ -210,6 +210,28 @@ async fn delete_worker(
         let delete_worker_message: DeleteMessage =
             serde_json::from_str(&serialized_message).expect("Failed to parse file message");
 
+        if delete_worker_message.empty_database {
+            match delete_chunks_in_dataset(
+                delete_worker_message.dataset_id,
+                web_pool.clone(),
+                delete_worker_message.server_config.clone(),
+            )
+            .await
+            {
+                Ok(_) => {
+                    log::info!(
+                        "Deleted all chunks for dataset: {:?}",
+                        delete_worker_message.dataset_id
+                    );
+                    continue;
+                }
+                Err(err) => {
+                    log::error!("Failed to delete all chunks for dataset: {:?}", err);
+                    continue;
+                }
```

Comment:
If the response is an `Ok` then it's important to remove it from the processing queue like: 

```
let _ = redis::cmd("LREM")
                    .arg("delete_dataset_processing")
                    .arg(1)
                    .arg(serialized_message)
                    .query_async::<redis::aio::MultiplexedConnection, usize>(&mut *redis_connection)
                    .await;
```

If the response is an error, then you need to add it back to the queue: 

```
let _ = readd_error_to_queue(delete_worker_message, err, redis_pool.clone()).await;
```

Reply:
Can reference the code slightly lower than this for how this should work.